fighters/packun/src/acmd/other.rs

```diff
@@ -128,21 +128,87 @@ unsafe fn appeal_hi_game(fighter: &mut L2CAgentBase) {
     let lua_state = fighter.lua_state_agent;
     let boma = fighter.boma();
     let cur_stance = VarModule::get_int(boma.object(), vars::packun::instance::CURRENT_STANCE);
-    VarModule::set_int(boma.object(), vars::packun::instance::CURRENT_STANCE, (cur_stance + 1) % 3);
-    VarModule::on_flag(fighter.object(), vars::packun::instance::STANCE_INIT);
-    frame(lua_state, 10.0);
+    let mut advance = 0;
+    let mut holding = true;
+    frame(lua_state, 1.0);
+    if is_excute(fighter) {
+        damage!(fighter, *MA_MSC_DAMAGE_DAMAGE_NO_REACTION, *DAMAGE_NO_REACTION_MODE_DAMAGE_POWER, 10.0);
+        if holding {
+            if !boma.is_button_on(Buttons::AppealAll) {
+                holding = false;
+            }
+        }
+    }
+    frame(lua_state, 2.0);
+    if is_excute(fighter) {
+        if holding {
+            if !boma.is_button_on(Buttons::AppealAll) {
+                holding = false;
+            }
+        }
+    }
+    frame(lua_state, 3.0);
+    if is_excute(fighter) {
+        if holding {
+            if !boma.is_button_on(Buttons::AppealAll) {
+                holding = false;
+            }
+        }
+    }
+    frame(lua_state, 4.0);
+    if is_excute(fighter) {
+        if holding {
+            if !boma.is_button_on(Buttons::AppealAll) {
+                holding = false;
+            }
+        }
+    }
+    frame(lua_state, 5.0);
+    if is_excute(fighter) {
+        if holding {
+            if !boma.is_button_on(Buttons::AppealAll) {
+                holding = false;
+            }
+        }
+    }
+    frame(lua_state, 6.0);
+    if is_excute(fighter) {
+        if holding {
+            if !boma.is_button_on(Buttons::AppealAll) {
+                holding = false;
+            }
+        }
+    }
+    frame(lua_state, 7.0);
+    if is_excute(fighter) {
+        if holding {
+            if !boma.is_button_on(Buttons::AppealAll) {
+                holding = false;
+            }
+        }
+    }
+    frame(lua_state, 8.0);
+    if is_excute(fighter) {
+        if holding {
+            advance = 2;
+        }
+        else {
+            advance = 1;
+        }
```

Comment:
since you only use the `advance` var here, please define it here too (instead of at the top of the function)

generally, you should only define variables as early as you need to reference them

Reply:
```rs
let advance = if holding {2} else {1};
```